framework/src/meshgenerators/BreakMeshByElementGenerator.C

```diff
@@ -25,17 +25,28 @@ BreakMeshByElementGenerator::validParams()
   InputParameters params = MeshGenerator::validParams();
   params.addClassDescription("Break all element-element interfaces in the specified subdomains.");
   params.addRequiredParam<MeshGeneratorName>("input", "The mesh we want to modify");
-  params.addParam<std::vector<SubdomainID>>("subdomains", "The list of subdomain IDs to explode.");
-  params.addRequiredParam<BoundaryName>(
-      "interface_name", "The boundary name containing all broken element-element interfaces.");
+  params.addParam<std::vector<SubdomainID>>(
+      "subdomains",
+      std::vector<SubdomainID>(),
+      "The list of subdomain IDs to explode.  Leave unset to explode all subdomains.");
+  params.addParam<BoundaryName>(
+      "interface_name",
+      "element_boundaries",
+      "The boundary name containing all broken element-element interfaces.");
+  params.addRangeCheckedParam<unsigned int>(
+      "interface_sides",
+      1,
+      "interface_sides<3",
+      "Whether to add no interface boundary, a 1-sided boundary, or a 2-sided boundary");
```

Comment:
explain the rules for 1 sided? is it from smaller id to biger elem id?

Reply:
I will describe the rule (which you're correct about), but I have no explanation for why anyone would have wanted that behavior.  :wink:  I'm just keeping it as the default and making more sensible behaviors an option under the principle that breaking existing behavior without gradual deprecation is rude.